pkg/driver/docker/docker.go

```diff
@@ -452,6 +489,83 @@ func (c dockerClient) waitUntilRunning(ctx context.Context,
 	}
 }
 
+func (c dockerClient) waitUntilRemoved(ctx context.Context,
+	name string, timeout time.Duration) error {
+	logger := logrus.WithField("container", name)
+	logger.Debug("waiting to be removed")
+
+	// Wait for the container to be removed
+	ctxTimeout, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	for {
+		select {
+		case <-time.After(waitingInterval):
+			exist, err := c.Exists(ctxTimeout, name)
+			if err != nil {
+				return errors.Wrap(err, "failed to check if container has been removed")
+			}
+			if !exist {
+				logger.Debug("the container has been removed")
+				return nil
+			}
+		case <-ctxTimeout.Done():
+			container, err := c.ContainerInspect(ctx, name)
+			if err != nil {
+				logger.Debugf("failed to inspect container %s", name)
+			}
+			state, err := json.Marshal(container.State)
+			if err != nil {
+				logger.Debug("failed to marshal container state")
+			}
+			logger.Debugf("container state: %s", state)
+			return errors.Errorf("timeout %s: container does not be removed", timeout)
+		}
+	}
+}
+
+func (c dockerClient) handleContainerCreated(ctx context.Context,
+	cname string, status containerType.ContainerStatus, timeout time.Duration) error {
+	logger := logrus.WithFields(logrus.Fields{
+		"container": cname,
+		"status":    status,
+	})
+
+	if status == containerType.StatusPaused {
+		logger.Info("container was paused, unpause it now...")
+		_, err := c.ResumeContainer(ctx, cname)
+		if err != nil {
+			return errors.Wrap(err, "failed to unpause container")
+		}
+	} else if status == containerType.StatusExited {
+		logger.Info("container exited, try to restart it...")
+		err := c.ContainerRestart(ctx, cname, container.StopOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to restart cotaniner")
+		}
+	} else if status == containerType.StatusDead {
+		logger.Info("container is dead, try to remove it...")
+		err := c.ContainerRemove(ctx, cname, types.ContainerRemoveOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to remove container")
+		}
+	} else if status == containerType.StatusRunning || status == containerType.StatusCreated {
+		logger.Info("container already exists.")
+		err := c.ContainerStart(ctx, cname, types.ContainerStartOptions{})
```

Comment:
What's the purpose of this action?

Reply:
Reserve the former logics... I'll remove them.